docs/development/testguide.rst

```diff
@@ -143,63 +160,65 @@ Astropy Test Function
 
 .. autofunction:: astropy.test
 
-pytest
-======
+tox
+===
 
-The test suite can be run directly from the native ``pytest`` command. In this
-case, it is important for developers to be aware that they must manually
-rebuild any extensions by running ``setup.py build_ext`` before testing.
+Finally, the most robust way to run the tests (which can also be the slowest) is
+to make use of `Tox <https://tox.readthedocs.io/en/latest/>`__, which is a
+general purpose tool for automating Python testing. One of the benefits of tox
+is that it first creates a source distribution of the package being tested, and
+installs it into a new virtual environment, along with any dependencies that are
+declared in the package, before running the tests. This can therefore catch
+issues related to undeclared package data, or missing dependencies. Since we use
+tox to run many of the tests on continuous integration services, it can also be
+used in many cases to reproduce issues seen on those services.
 
-In contrast to the case of running from ``setup.py``, the ``--doctest-plus``
-and ``--doctest-rst`` options are not enabled by default when running the
-``pytest`` command directly. This flags should be explicitly given if they are
-needed.
+To run the tests with tox, first make sure that tox is installed, e.g.::
 
-Test-running options
-====================
+    pip install tox
 
-Running parts of the test suite
--------------------------------
+then run the basic test suite with::
 
-It is possible to run only the tests for a particular subpackage or set of
-subpackages.  For example, to run only the ``wcs`` tests from the
-commandline::
+    tox -e test
 
-    python setup.py test -P wcs
+or run the test suite with all optional dependencies with::
 
-Or, to run only the ``wcs`` and ``utils`` tests::
+    tox -e test-all
```

Comment:
I think in `.travis.yml` this was `test-all-deps`....

Reply:
Fixed